packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1536,26 +1536,40 @@ export class MergeTree {
 				const starts = new Set<ObliterateInfo>();
 				const overlapping = new Set<ObliterateInfo>();
 				const findLeftMovedSegment = (seg: ISegment): boolean => {
-					seg.localRefs?.walkReferences(
-						(ref) => {
-							const obliterate = ref.properties?.obliterate as ObliterateInfo | undefined;
-							if (obliterate?.start === ref) {
-								starts.add(obliterate);
-							}
-						},
-						undefined,
-						false,
-					);
+					if (
+						(seg.movedSeq !== undefined &&
+							seg.movedSeq !== UnassignedSequenceNumber &&
+							this.moveSeqs.includes(seg.movedSeq)) ||
+						(seg.localMovedSeq !== undefined && this.localMoveSeqs.has(seg.localMovedSeq))
+					) {
+						seg.localRefs?.walkReferences(
```

Comment:
i think it would be safe to only check the first and last position of the segment, rather than scan the whole thing, as we shouldn't merge segments above the min seq, so the obliterate refs should always stay on the ends of the segments